src/spatial/models/report_datafetcher.py

```diff
@@ -0,0 +1,214 @@
+import requests
+import openai
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from huggingface_hub import login
+from configure import Config
+import google.generativeai as genai 
+
+
+# Configure API keys
+GOOGLE_API_KEY = Config.GOOGLE_API_KEY
+genai.configure(api_key=GOOGLE_API_KEY)
+hf_token = Config.HUGGING_FACE_TOKEN
+
+if hf_token:
+    login(hf_token)
+else:
+    print("Hugging Face token is missing. Set the 'HUGGING_FACE_TOKEN' environment variable.")
+
+class DataFetcher:
+    @staticmethod
+    def fetch_air_quality_data_a(grid_id, start_time, end_time):
+        token = Config.AIRQO_API_TOKEN  
+        if token is None:
+            print("Error: AIRQO_API_TOKEN environment variable is not set.")
+            return None
+
+        url = f"https://platform.airqo.net/api/v2/analytics/grid/report?token={token}"
```

Comment:
@wabinyai , is it possible to have this URL (or a section of it) as an environmental variable? Thanks for the great work.

Reply:
@Baalmart I am still doing some extra work on it. 
The environment needed are **HUGGING_FACE_TOKEN**  **GOOGLE_API_KEY**   **OPENAI_API_KEY**
We are still exploring  other free LLM that can do as well as gemini